autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/FlowCommunicator.java

```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.periscope.monitor.handler;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Lists;
+import com.sequenceiq.cloudbreak.client.CloudbreakInternalCrnClient;
+import com.sequenceiq.flow.api.model.FlowCheckResponse;
+
+@Service
+public class FlowCommunicator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowCommunicator.class);
+
+    private static final Integer PAGE_SIZE = 50;
+
+    private static final Integer PAGE = 0;
+
+    @Inject
+    private CloudbreakInternalCrnClient internalCrnClient;
+
+    public Map<Long, FlowCheckResponse> getFlowStatusFromFlowIds(Map<String, Long> activityIdsByFlowIds) {
+        LOGGER.info("Sending request to fetch FlowStatus of {} flowIds", activityIdsByFlowIds.size());
+        List<String> allFlowIds = newArrayList(activityIdsByFlowIds.keySet());
+        List<FlowCheckResponse> flowCheckResponses = newArrayList();
+        Lists.partition(allFlowIds, PAGE_SIZE).forEach(flowIds ->
+                flowCheckResponses.addAll(internalCrnClient.withInternalCrn().flowEndpoint().getFlowChainsStatusesByChainIds(flowIds, flowIds.size(), PAGE)));
```

Comment:
A few questions on this.
1) Is this a parallel call?
2) What happens if one of the invocation fails - will we have to go after all of the requestIds again?
3) What happens if Cloudbreak, for some reason, does not have a specific flowId registered - what does the response look like? (Do we have a unit test which exercises this scenario)

Reply:
> 1. Is this a parallel call?
> 2. What happens if one of the invocation fails - will we have to go after all of the requestIds again?

This is a sequential call. And we will likely have to query the same list flowIds again in case an exception is encountered, since their flow status will not have been updated.

> 3.What happens if Cloudbreak, for some reason, does not have a specific flowId registered - what does the response look like? (Do we have a unit test which exercises this scenario)

Not in periscope. A UT will have to be added to exercise this scenario. Do we possibly add a new state for such records - something along the lines of `SCALING_FLOW_UNKNOWN`?